From a23466ccf3bc912583491f587115abacff5edfc6 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:05 -0500
Subject: [drm] nouveau/bar: behave better if ioremap failed

Message-id: <1418852380-13061-55-git-send-email-rclark@redhat.com>
Patchwork-id: 102191
O-Subject: [RHEL7 drm 54/89] drm/nouveau/bar: behave better if ioremap failed
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: 3.17

commit f011ba5cafbae6e2d844e16d98b210d41d6e57c8

Author:     Ben Skeggs <bskeggs@redhat.com>
AuthorDate: Mon Aug 11 11:52:44 2014 +1000
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:45:28 2014 -0500

    drm/nouveau/bar: behave better if ioremap failed

    commit e7d96929a7cd93e146af0d1fd251fa1ec2e3ebf8 upstream

    Signed-off-by: Ben Skeggs <bskeggs@redhat.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/nouveau/core/subdev/bar/base.c b/drivers/gpu/drm/nouveau/core/subdev/bar/base.c
index 73b1ed2..8bcbdf3 100644
--- a/drivers/gpu/drm/nouveau/core/subdev/bar/base.c
+++ b/drivers/gpu/drm/nouveau/core/subdev/bar/base.c
@@ -99,8 +99,13 @@ nouveau_bar_alloc(struct nouveau_bar *bar, struct nouveau_object *parent,
     struct nouveau_mem *mem, struct nouveau_object **pobject)
 {
  struct nouveau_object *engine = nv_object(bar);
- return nouveau_object_ctor(parent, engine, &nouveau_barobj_oclass,
-       mem, 0, pobject);
+ int ret = -ENOMEM;
+ if (bar->iomem) {
+  ret = nouveau_object_ctor(parent, engine,
+       &nouveau_barobj_oclass,
+       mem, 0, pobject);
+ }
+ return ret;
 }
 
 int
@@ -118,9 +123,12 @@ nouveau_bar_create_(struct nouveau_object *parent,
  if (ret)
   return ret;
 
- if (nv_device_resource_len(device, 3) != 0)
+ if (nv_device_resource_len(device, 3) != 0) {
   bar->iomem = ioremap(nv_device_resource_start(device, 3),
          nv_device_resource_len(device, 3));
+  if (!bar->iomem)
+   nv_warn(bar, "PRAMIN ioremap failed\n");
+ }
 
  return 0;
 }
-- 
1.7.1